REGRESSION(294925@main): [SwiftUI][macOS] Find navigator is misplaced

jesxilin opened the pull request:
#### 57e000f31b6d8f4e34faf588a8204c6f87242566
<pre>
REGRESSION(294925@main): [SwiftUI][macOS] Find navigator is misplaced
<a href="https://bugs.webkit.org/show_bug.cgi?id=293638">https://bugs.webkit.org/show_bug.cgi?id=293638</a>
<a href="https://rdar.apple.com/152109281">rdar://152109281</a>

Reviewed by Richard Robinson.

Guard ignoresSafeArea in WebView to be used only for iOS
since
1) always calling ignoresSafeArea on macOS causes
Find Navigator to disrespect safe areas
2) the call to ignoresSafeArea was meant for iOS.

Refactor logic to a var representable.

* Source/WebKit/_WebKit_SwiftUI/API/WebView.swift:
(WebView.representable):
(WebView.body):

Canonical link: <a href="https://commits.webkit.org/295517@main">https://commits.webkit.org/295517@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/db865b3543480bae363a89a42967013a46de6b28

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/105346 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/131/builds/25058 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/15485 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/110554 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/56001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/130/builds/25496 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/123/builds/33601 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/110554 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/56001 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108352 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/130/builds/25496 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/95082 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/110554 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/130/builds/25496 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/55398 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/130/builds/25496 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/13204 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/113214 "Built successfully") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/128/builds/32501 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/123/builds/33601 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/89088 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/121/builds/32864 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/91302 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/88729 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/11414 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/27974 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17080 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/127/builds/32425 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/37840 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/32195 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/35541 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/33771 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1d2a1c1ca3f699ca194d49f79f0ded8fe1d60a50. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45983

pxlcoder commented on Source/WebKit/_WebKit_SwiftUI/API/WebView.swift:
```diff
@@ -67,19 +67,29 @@ public struct WebView: View {
 
-    // swift-format-ignore: AllPublicDeclarationsHaveDocumentation
-    public var body: some View {
+    @ViewBuilder
```
For my own understanding, do we need `@ViewBuilder` here because it returns a different value depending on the platform?

Otherwise, I thought it was only needed when returning a tuple of Views.

rr-codes commented on Source/WebKit/_WebKit_SwiftUI/API/WebView.swift:
```diff
@@ -67,19 +67,29 @@ public struct WebView: View {
-    // swift-format-ignore: AllPublicDeclarationsHaveDocumentation
-    public var body: some View {
+    @ViewBuilder
+    private var webViewRepresentableWithSafeArea: some View {
```
I think this should be named like `contents` or something because (a) the "withSafeArea" part is a bit confusing because it is also _ignoring_ the safe area in the implementation and (b) the fact that its a webViewRepresentable isn't really that important to the name

jesxilin replied:
I was thinking since the original line would've been `webViewRepresentable`... I'd keep webViewRepresentable. Is `contents` a type of swiftUI naming convention?

rr-codes replied:
`contents` isn't a swiftui convention, was just a suggestion. `webViewRepresentable` or `representable` are fine!

jesxilin commented on Source/WebKit/_WebKit_SwiftUI/API/WebView.swift:
```diff
@@ -67,19 +67,29 @@ public struct WebView: View {
                 .ignoresSafeArea()
         }
+        #else
+        let zeroEdgeInsets = EdgeInsets()
+        WebViewRepresentable(page: storage.webPage, safeAreaInsets: zeroEdgeInsets)
```
Will do!

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/db865b3543480bae363a89a42967013a46de6b28. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/45983

jesxilin committed:
REGRESSION(294925@main): [SwiftUI][macOS] Find navigator is misplaced
https://bugs.webkit.org/show_bug.cgi?id=293638
rdar://152109281

Reviewed by Richard Robinson.

Guard ignoresSafeArea in WebView to be used only for iOS
since
1) always calling ignoresSafeArea on macOS causes
Find Navigator to disrespect safe areas
2) the call to ignoresSafeArea was meant for iOS.

Refactor logic to a var representable.

* Source/WebKit/_WebKit_SwiftUI/API/WebView.swift:
(WebView.representable):
(WebView.body):

Canonical link: https://commits.webkit.org/295517@main
